WordPress/gutenberg
# Font Library: run wp_font_family post content sanitization using wp_insert_post_data filter

## What?
This PR changes where and when data sanitization happens, running it right before it is inserted into the database using the `wp_insert_post_data` filter.

## Why?
This is done for security reasons, to ensure that we are not storing data that is unescaped. 
The `wp_font_family` posts could theoretically be created outside of the context of the Font Library class, e.g. via manual insertion of a post, and currently we are only calling the sanitization method inside of the Font Library class. 

## How?
Calling the sanitization function on `wp_font_family` type posts to filter the data before insertion into the database.

## Testing Instructions
1. Try to save a wp_font_family post with inappropriate content.
2. Check if the data was sanitized.

~Marking this as a draft because it requires https://github.com/WordPress/gutenberg/pull/53273 to work as expected.~
